src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -208,6 +209,42 @@ get_tiles(const pt::ptree& aie_meta, const std::string& graph_name)
   return tiles;
 }
 
+std::vector<tile_type>
+get_event_tiles(const pt::ptree& aie_meta, const std::string& graph_name,
+                e_tile_type type)
+{
+  std::vector<tile_type> tiles;
+
+  // Not supported yet
+  if (type == xrt_core::edge::aie::AIE_TILE_SHIM)
+    return tiles;
+
+  const char* key_cols[xrt_core::edge::aie::AIE_TILE_NUM_TYPES] =
+      {"core_columns", "dma_columns", "N/A"};
+  const char* key_rows[xrt_core::edge::aie::AIE_TILE_NUM_TYPES] =
+      {"core_rows", "dma_rows", "N/A"};
+
+  for (auto& graph : aie_meta.get_child("aie_metadata.EventGraphs")) {
+    if (graph.second.get<std::string>("name") != graph_name)
+      continue;
+
+    int count = 0;
+      for (auto& node : graph.second.get_child( key_cols[type] )) {
```

Comment:
When type becomes an enum class (see later), you will have to` static_cast` the type to integral.  While that looks cumbersome, it is in fact better coding style since your intend is more clearly stated.  For example as is right now, I could be let to think that using `type` as array index is a bug, but if you clearly static_cast it to integral, then it is expressed clearly what the intend is.